.pfnci/run.sh

```diff
@@ -0,0 +1,187 @@
+#       testing.
+# - SPREADSHEET_ID ... Set SPREADSHEET_ID (e.g.,
+#       "1u5OYiPOL3XRppn73XBSgR-XyDuHKb_4Ilmx1kgJfa-k") to enable xpytest to
+#       report to a spreadsheet.
+
+set -eux
+
+cp -a /src /chainer
+cp /chainer/setup.cfg /
+cd /
+
+# Remove pyc files.  When the CI is triggered with a user's local files, pyc
+# files generated on the user's local machine and they often cause failures.
+find /chainer -name "*.pyc" -exec rm -f {} \;
+
+TARGET="$1"
+: "${GPU:=0}"
+: "${XPYTEST_NUM_THREADS:=$(nproc)}"
+
+################################################################################
+# Test functions
+################################################################################
+
+# test_py37 is a test function for chainer.py37.
+test_py37() {
+  #-----------------------------------------------------------------------------
+  # Configure parameters
+  #-----------------------------------------------------------------------------
+  marker='not slow'
+  if (( !GPU )); then
+    marker+=' and not gpu and not cudnn'
+    bucket=1
+    export CHAINERX_TEST_CUDA_DEVICE_LIMIT=0
+  else
+    marker+=' and (gpu or cudnn)'
+    bucket="${GPU}"
+  fi
+
+  #-----------------------------------------------------------------------------
+  # Install Chainer
+  #-----------------------------------------------------------------------------
+  CHAINER_BUILD_CHAINERX=1 CHAINERX_BUILD_CUDA=1 MAKEFLAGS="-j$(nproc)" \
+  CHAINERX_NVCC_GENERATE_CODE=arch=compute_70,code=sm_70 \
+      python3.7 -m pip install /chainer[test] 2>&1 >/tmp/install.log &
+  install_pid=$!
+
+  #-----------------------------------------------------------------------------
+  # ChainerX C++ tests
+  #-----------------------------------------------------------------------------
+  mkdir -p /root/chainerx_build
+  pushd /root/chainerx_build
+  cmake \
+      -DCMAKE_BUILD_TYPE=Debug \
+      -DCHAINERX_BUILD_CUDA="$(
+          if (( GPU )); then echo 'ON'; else echo 'OFF'; fi
+      )" \
+      -DCHAINERX_BUILD_TEST=ON \
+      -DCHAINERX_BUILD_PYTHON=OFF \
+      -DCHAINERX_WARNINGS_AS_ERRORS=ON \
+      /chainer/chainerx_cc
+  # NOTE: Use nice to prioritize pip install process.
+  nice -n 19 make "-j$(nproc)"
+  ctest --output-on-failure "-j$(nproc)" && :
+  cc_test_status=$?
+  popd
+
+  #-----------------------------------------------------------------------------
+  # Chainer/ChainerX python tests
+  #-----------------------------------------------------------------------------
+  if ! wait $install_pid; then
+    cat /tmp/install.log
+    exit 1
+  fi
+  xpytest_args=(
+      --python=python3.7 -m "${marker}"
+      --bucket="${bucket}" --thread="$(( XPYTEST_NUM_THREADS / bucket ))"
+      --hint="/chainer/.pfnci/hint.pbtxt"
+  )
+  if [ "${SPREADSHEET_ID:-}" != '' ]; then
+    xpytest_args+=(--spreadsheet_id="${SPREADSHEET_ID}")
+  fi
+  OMP_NUM_THREADS=1 xpytest "${xpytest_args[@]}" \
+      '/chainer/tests/chainerx_tests/**/test_*.py' \
```

Comment:
Sorry, but this is required for now because xpytest passes a file name to pytest explicitly and passing __init__.py will cause an error even if setup.cfg is read by pytest.

Reply:
By the way, do you have a plan to support `setup.cfg` anytime soon? Obviously it's not great to maintain file name patterns in multiple places.

Reply:
Note that this supports `setup.cfg` features other than file selection features such as `testpaths` and `python_files`. They can be supported by `xpytest` if you want. I added a TODO to make xpytest support the feature.

Reply:
OK.
I hope this duplication will be eventually resolved.
Although I think it's better to use pytest's own customization mechanisms to distribute tests, it's out of scope of this PR.